backends/ref/ceed-ref-operator.c

```diff
@@ -1171,27 +1194,33 @@ int CeedOperatorCreateFDMElementInverse_Ref(CeedOperator op,
 
   // Build FDM diagonal
   CeedVector q_data;
-  CeedScalar *q_data_array;
-  ierr = CeedVectorCreate(ceed_parent, num_elem*num_comp*l_size, &q_data);
+  CeedScalar *q_data_array, *fdm_diagonal;
+  ierr = CeedCalloc(num_comp*elem_size, &fdm_diagonal); CeedChkBackend(ierr);
+  for (CeedInt c=0; c<num_comp; c++)
+    for (CeedInt n=0; n<elem_size; n++) {
+      if (interp)
+        fdm_diagonal[c*elem_size + n] = 1.0;
+      if (grad)
+        for (CeedInt d=0; d<dim; d++) {
+          CeedInt i = (n / CeedIntPow(P_1d, d)) % P_1d;
+          fdm_diagonal[c*elem_size + n] += lambda[i];
+        }
+      if (fabs(fdm_diagonal[c*elem_size + n]) < 1000*CEED_EPSILON)
+        fdm_diagonal[c*elem_size + n] = 1000*CEED_EPSILON;
```

Comment:
This is internal and for a specific purpose, but I don't love thresholding based on absolute values instead of relative values. Changing the entries based on choice of units or mesh resolution shouldn't cause qualitatively different local behavior. Thresholding could be based on max value or some other relative quantity.